tests/workflows/test_consistency.py

```diff
@@ -0,0 +1,216 @@
+            grand_mean_scaling=dict(mean=10000.0),
+            ica_aroma=False,
+        ),
+    ),
+    TestSetting(
+        name="simple",  # was FalseComb3
+        base_setting=dict(
+            confounds_removal=[
+                "(trans|rot)_[xyz]",
+                "(trans|rot)_[xyz]_derivative1",
+                "(trans|rot)_[xyz]_power2",
+                "(trans|rot)_[xyz]_derivative1_power2",
+            ],
+            grand_mean_scaling=dict(mean=10000.0),
+            ica_aroma=False,
+        ),
+    ),
+    TestSetting(
+        name="simpleGSR",  # was FalseComb4
+        base_setting=dict(
+            confounds_removal=[
+                "(trans|rot)_[xyz]",
+                "(trans|rot)_[xyz]_derivative1",
+                "(trans|rot)_[xyz]_power2",
+                "(trans|rot)_[xyz]_derivative1_power2",
+                "global_signal",
+            ],
+            grand_mean_scaling=dict(mean=10000.0),
+            ica_aroma=False,
+        ),
+    ),
+    TestSetting(
+        name="GSR",  # was FalseComb5
+        base_setting=dict(
+            confounds_removal=[
+                "global_signal",
+            ],
+            grand_mean_scaling=dict(mean=10000.0),
+            ica_aroma=False,
+        ),
+    ),
+]
+
+
+@pytest.mark.parametrize("dataset", datasets)
+def test_extraction(dataset: Dataset, tmp_path: Path, pcc_mask: Path):
+    """
+    Run preprocessing and feature extraction for each of the participants coming the list of datasets,
+    and each of the settings coming from settings_list. Standard TSNR is calculated inside thefmriprep workflows.
+    We create a zip file with all the features, tsnr and spec file, per participant.
+    """
+
+    dataset_file = dataset.download(tmp_path)
+
+    spec = make_spec(dataset_files=[dataset_file], pcc_mask=pcc_mask, test_settings=settings_list)
+    config.nipype.omp_nthreads = cpu_count()
+    save_spec(spec, workdir=tmp_path)
+
+    # Run built-in halfpipe
+    workflow = init_workflow(tmp_path)
+    graphs = init_execgraph(tmp_path, workflow)
+    parser = build_parser()
+    opts = parser.parse_args(args=list())
+    opts.graphs = graphs
+    opts.debug = True
+    opts.nipype_run_plugin = "MultiProc"  # "Simple", "MultiProc"
+    opts.workdir = tmp_path  # necessary when you choose MultiProc
+
+    run_stage_run(opts)
+    # OR: Run another halfpipe version (not implemented)
+    # subprocess.call(["docker", "run", "--volume", f"{tmp_path}:{tmp_path}", "--rm", "halfpipe/halfpipe:1.2.2 "])
+
+    index = BIDSIndex()
+    index.put(tmp_path / "derivatives")
+    spec_file = tmp_path / "spec.json"
+
+    for sub in dataset.subject_ids:
+        paths_to_zip = []
+        for test_setting in settings_list:
+            name = test_setting.name
+
+            for title, kwargs in [
+                ("Timeseries", dict(sub=sub, feature=f"{name}CorrMatrix", suffix="timeseries", task="rest", extension=".tsv")),
```

Comment:
Did we also want to add task-based here?

Reply:
I thought we decided in the meetings that we would not have a task-based feature. Also the subjects come from different datasets so it would mean different contrasts and conditions depending on what was the task

Reply:
Hmm, I remember discussing the possibility of using fake task events (which would be the same for all datasets, and are already implemented for the unit tests). I guess we can always add that later

Reply:
I think maybe we can re-discuss this then and see if we want to add it in the end :), but for this PR Id say we can leave it (also the 100 iterations I already ran did not contain task-based features)